src/Objects/MonitorManager.vala

```diff
@@ -188,6 +198,7 @@ public class Display.MonitorManager : GLib.Object {
             virtual_monitor.current_x = mutter_logical_monitor.x;
             virtual_monitor.current_y = mutter_logical_monitor.y;
             virtual_monitor.scale = mutter_logical_monitor.scale;
+            max_scale = virtual_monitor.scale;
```

Comment:
Did you mean this? Or `max_scale = double.min (max_scale, virtual_monitor.scale);`